### ISSUE TITLE
Applying recipes for push/append/setindex

### ISSUE THREAD

**Issue by tbreloff** [tbreloff/Plots.jl#327]
Moving from https://github.com/JuliaPlots/RecipesBase.jl/issues/8.

Some serious thought needs to go into how this might work, but the idea is that updating the series data with `push!` or `setindex!` would first apply applicable recipes before updating the data.  To get this right, we would need to store: 

- a history of transformations to apply to the data
- when multiple series are created, the series that the final data subset should be mapped to.

As an example, if we push to a `histogram` series, then it should know that the transformations (recipes) were "histogram --> bar --> filled path".  We'd need to update the original data with a push/set and then recompute the transformations.

In short, this is doable and very powerful, but needs some real thought and some additional machinery to get it right.

Brainstorm: can we simply store the original args/kw with the series, update the args with the values from the push/append/setindex call, and then re-generate the series?  Then we might be able to generalize to a `update!(series::Series, args...; kw...)`, and might solve things like https://github.com/tbreloff/Plots.jl/issues/62 in the process.

cc: @joshday
